File: twoSum.cpp
#include <iostream>
#include <set>
using namespace std;

bool twoSum(int arr[],int n,int target)
{
    set<int>hash_set;
    for(int i{};i<n;i++)
    {
        if(hash_set.find(target-arr[i])!=hash_set.end())
            return true;
        hash_set.insert(arr[i]);
    }
    return false;
}

int main()
{
    int n;
    cin>>n;
    int target;
    cin>>target;
    int arr[n];
    for(int i{};i<n;i++)
        cin>>arr[i];
    cout<<twoSum(arr,n,target)<<endl;
}

File: twoSum.py
def two_sum(list, target):
    s=set()
    for i in list:
        if target-i in s:
            return True
        s.add(i)
    return False
    

print(two_sum([4,7,1,-3,2], 5)) #True
print(two_sum([4,7,1,-3,2],10)) #False
